File: coding/19th January 2025/Group Anagrams.java
//Group Anagrams
class Solution 
{
    public List<List<String>> groupAnagrams(String[] strs) 
    {
        HashMap<String, List<String>> res = new HashMap<>();
        for (String s : strs)
        {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String schars = new String(chars);
            if(!res.containsKey(schars))
                res.put(schars, new ArrayList<>());
            res.get(schars).add(s);
        }
        ArrayList result = new ArrayList<>(res.values());
        return result;
    }
}

File: Coding in Python/08th January 2026/Group Anagrams.py
#Group Anagrams
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        result = []
        d = dict()
        for s in strs:
            st = ''.join(sorted(s))
            if(st not in d):
                d[st] = []
            d[st].append(s)
        for k in d:
            result.append(d[k])
        return result
